sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala

```diff
@@ -31,13 +32,26 @@ class SparkPlanInfo(
     val simpleString: String,
     val children: Seq[SparkPlanInfo],
     val metadata: Map[String, String],
-    val metrics: Seq[SQLMetricInfo])
+    val metrics: Seq[SQLMetricInfo]) {
+
+  override def hashCode(): Int = {
+    simpleString.hashCode
```

Comment:
We only need to make sure that those object equal with each other should have the same hashCode, but the hashCode does not need to considering all the members.

Using simpleString here should be enough to have good hashCode (be good enough to distinguish each other within a plan)